torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -2514,6 +2516,76 @@ def call_function(
         return _make_inlined(tx, pytree.tree_unflatten)(variable, treespec)
 
 
+# Copy paste of the above class, but enables an arbitrary callable
+class WrapGenericHigherOrderVariable(WrapHigherOrderVariable):
+    def __init__(self, hop, source) -> None:
+        super().__init__(hop, source)
+
+    def call_function(
+        self,
+        tx: "InstructionTranslator",
+        args: list[VariableTracker],
+        kwargs: dict[str, VariableTracker],
+    ) -> VariableTracker:
+        from .builder import wrap_fx_proxy
+
+        func_var = kwargs["wrapper_fn"]
+
+        if isinstance(func_var, torch._dynamo.variables.UserFunctionVariable):
+            func = func_var.fn
+        elif isinstance(
+            func_var, torch._dynamo.variables.functions.FunctoolsPartialVariable
+        ):
+            func = func_var.as_python_constant()
+        else:
+            raise RuntimeError(
+                f"WrapGenericHigherOrderVariable: Unsupported function {type(func_var)}"
+            )
+        gmod_kwargs = {
```

Comment:
Using the wrapper_fn as kwargs seems pretty fragile like it might collide with inner kwargs. Maybe we could just pass the wrapper fn as an explicit input? Something like:
```
DynamoBypassingWrapper(wrapper_fn, inner_fn, *inner_args, **inner_kwargs)
```

The proxy node we created in dynamo could make wrapper_fn a string key that can be used to access gmod.meta.

Reply:
Good point, updated